core/metacling/src/TClingCallFunc.cxx

```diff
@@ -1998,11 +1998,15 @@ TInterpreter::CallFuncIFacePtr_t TClingCallFunc::IFacePtr()
       const Decl *decl = GetFunctionOrShadowDecl();
 
       R__LOCKGUARD_CLING(gInterpreterMutex);
-      map<const Decl *, void *>::iterator I = gWrapperStore.find(decl);
-      if (I != gWrapperStore.end()) {
-         fWrapper = (tcling_callfunc_Wrapper_t) I->second;
-      } else {
-         fWrapper = make_wrapper();
+
+      // check if another thread already did it
+      if (!fWrapper) {
```

Comment:
With this pattern (same with the previous one :) ), `fWrapper` actually should be an atomic pointer.  I.e. another thread might be executing line 1997 as the same time as this thread execute line 2006 (or 2008) and might get a partial value.   Could you update `fWrapper` to become an atomic?